src/vivarium/component.py

```diff
@@ -671,32 +673,49 @@ def get_data(
         layered_config_tree.exceptions.ConfigurationError
             If the data source is invalid.
         """
-        if isinstance(data_source, (float, int, list, pd.DataFrame)):
-            return data_source
-
-        if "::" in data_source:
-            module, method = data_source.split("::")
-            try:
-                if module == "self":
-                    data_getter = getattr(self, method)
-                else:
-                    data_getter = getattr(import_module(module), method)
-            except ModuleNotFoundError:
-                raise ConfigurationError(f"Unable to find module '{module}'.")
-            except AttributeError:
-                module_string = (
-                    f"component {self.name}." if module == "self" else f"module '{module}'."
-                )
-                raise ConfigurationError(
-                    f"There is no method '{method}' for the {module_string}."
-                )
-
-            return data_getter(builder)
-
-        try:
-            return builder.data.load(data_source)
-        except ArtifactException:
-            raise ConfigurationError(f"Failed to find key '{data_source}' in artifact.")
+        # TODO update this to use vivarium.types.LookupTableData once we drop
```

Comment:
Is there a ticket for this?

Reply:
I'm not sure we have a ticket, but we've discussed that we want to do it sometime in the next six months